diff --git a/sound/usb/quirks.c b/sound/usb/quirks.c
index 58b37bfc885c..85fe629cf1b5 100644
--- a/sound/usb/quirks.c
+++ b/sound/usb/quirks.c
@@ -2174,6 +2174,8 @@ static const struct usb_audio_quirk_flags_table quirk_flags_table[] = {
 		   QUIRK_FLAG_DSD_RAW),
 	VENDOR_FLG(0x22d9, /* Oppo */
 		   QUIRK_FLAG_DSD_RAW),
+    VENDOR_FLG(0x22e8, /* Cambridge */
+           QUIRK_FLAG_DSD_RAW),
 	VENDOR_FLG(0x23ba, /* Playback Design */
 		   QUIRK_FLAG_CTL_MSG_DELAY | QUIRK_FLAG_IFACE_DELAY |
 		   QUIRK_FLAG_DSD_RAW),
